src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobTask.cs

```diff
@@ -106,46 +113,40 @@ public async Task ExecuteAsync(ExportJobRecord exportJobRecord, WeakETag weakETa
                 // 2. There is no continuation token but the page is 0, which means it's the initial export.
                 while (progress.ContinuationToken != null || progress.Page == 0)
                 {
-                    // Commit the changes if necessary.
-                    if (progress.Page != 0 && progress.Page % _exportJobConfiguration.NumberOfPagesPerCommit == 0)
-                    {
-                        await _exportDestinationClient.CommitAsync(cancellationToken);
-
-                        // Update the job record.
-                        await UpdateJobRecord(_exportJobRecord, cancellationToken);
-
-                        currentBatchId = progress.Page;
-                    }
-
-                    // Set the continuation token.
-                    queryParameters[0] = Tuple.Create(KnownQueryParameterNames.ContinuationToken, progress.ContinuationToken);
-
+                    // Search and process the results.
                     SearchResult searchResult = await _searchService.SearchAsync(_exportJobRecord.ResourceType, queryParameters, cancellationToken);
-
-                    foreach (ResourceWrapper resourceWrapper in searchResult.Results)
-                    {
-                        await ProcessResourceWrapperAsync(resourceWrapper, currentBatchId, cancellationToken);
-                    }
+                    await ProcessSearchResults(searchResult.Results, currentBatchId, cancellationToken);
 
                     if (searchResult.ContinuationToken == null)
                     {
                         // No more continuation token, we are done.
                         break;
```

Comment:
nit: the `progress.Page != 0` below in the if statement can be removed now since you changed the order of execution.